internal/rotate.go

```diff
@@ -93,6 +112,10 @@ func (h *CertRotationHandler) Save() error {
 }
 
 func (h *CertRotationHandler) Rotate() error {
+	var err error
+	defer h.eventSync.NotifyCompleted(err)
+	h.eventSync.NotifyStarted()
+
 	// Before we even start - we should save our initial state (ie EstServer)
 	// and also make sure we *can* save our state.
 	if err := h.Save(); err != nil {
```

Comment:
This `err` also overrides the top-level `err` (i.e. is not used in the `NotifyCompleted` call).

Reply:
ea2d92f6e660149d175107397ce4ecdbbd2b90aa